### PR TITLE
#1283 Add revalidate method

### LINKED ISSUE #1283: Provide method to force revalidation
It would be nice to have a method on `ValidationSupport` to force re-validation. Currently, the only way that validation is triggered is when the value changes. There are cases where the validity of a control can change even if the value of that control does not.

### PR DESCRIPTION
**Pull request by shollander**
Fixes #1283 for master (java 8)

### CODE REVIEW

File: controlsfx/src/main/java/org/controlsfx/validation/ValidationSupport.java [hunk 1]
```diff
@@ -158,6 +158,24 @@ public void redecorate() {
...
         }
     }
+
+    /**
+     * Triggers validation for all known components.
+     * It is only necessary to call this if it is needed to revalidate even if the value of the control has not changed.
+     */
+    public void revalidate() {
+        for (Control target : getRegisteredControls()) {
+            target.fireEvent(new ValidateEvent());
```
Comment (eugener) [inline controlsfx/src/main/java/org/controlsfx/validation/ValidationSupport.java:168]: I wonder why ValidationEvent instance has to be created every time? 
It does not cary any specific information. I'd think singleton instance should be enough.

File: controlsfx/src/main/java/org/controlsfx/validation/ValidationSupport.java [hunk 2]
```diff
@@ -158,6 +158,24 @@ public void redecorate() {
...
+        }
+    }
+
+    /**
+     * Triggers validation for the given component.
+     * It is only necessary to call this if it is needed to revalidate even if the value of the control has not changed.
+     */
+    public void revalidate(Control c) {
+        c.fireEvent(new ValidateEvent());
```
Comment (eugener) [inline controlsfx/src/main/java/org/controlsfx/validation/ValidationSupport.java:177]: Same here.... singleton instance can be used

File: controlsfx/src/main/java/org/controlsfx/validation/ValidationSupport.java (hunk 1 above)
Comment (shollander) [reply controlsfx/src/main/java/org/controlsfx/validation/ValidationSupport.java:168]: Event objects are not meant to be reused. If you look at the parent `javafx.event.Event` class it does carry state such as the `consumed` field. Logically also, it is a different occurrence of the event, so it should be a different object.

Comment (eugener) [reply controlsfx/src/main/java/org/controlsfx/validation/ValidationSupport.java:168]: Not a big deal... was simply wondering if optimization is possible.

**Pull request merged**